contracts/staking/ServiceStakingBase.sol

```diff
@@ -457,9 +523,16 @@ abstract contract ServiceStakingBase is ERC721TokenReceiver, IErrorsRegistries {
             uint256[] memory serviceIds, uint256[][] memory serviceNonces,
             uint256[] memory serviceInactivity) = _calculateStakingRewards();
 
+        // Get arrays for eligible service Ids and rewards of exact size
+        uint256[] memory finalEligibleServiceIds;
+        uint256[] memory finalEligibleServiceRewards;
+        evictServiceIds = new uint256[](serviceIds.length);
+        uint256 curServiceId;
```

Comment:
For the event, we need a precise number of eligible services.

Also, evict services are allocated to the size of the all services in the set of staked services.